### PR TITLE
Code Ready to be tested for UNH Release code

### PR DESCRIPTION
**Pull request by atrimper**
- See commits for change notes
- Untested auto selector
- Untested spin speed
- Untested spin up shooter at end of loading process

### CODE REVIEW

File: Vision/vision2016.py [hunk 1]
```diff
@@ -39,13 +39,13 @@ def findDistanceToTarget(width):
     return distance
 
 def findAngle(distance):
-    angle = math.degrees(math.atan(7/distance))
+    angle = (.1183*(distance **2 ) - (3.468 * distance) + 69.203)
```
Comment (msoucy) [inline Vision/vision2016.py:42]: Ideally leave a comment here explaining that the values came from testing, or what they mean, etc.

File: src/org/usfirst/frc/team166/robot/commands/LoadingProcess.java [hunk 2]
```diff
@@ -21,5 +23,6 @@ public LoadingProcess() {
 		addSequential(new RollerSequence());
 		addSequential(new IntakeMotorStop());
 		addSequential(new RaiseRake());
+		addSequential(new SetShooterSpeed(Preferences.getInstance().getDouble("ShooterSpeed", 0.0)));
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/commands/LoadingProcess.java:26]: Why does it default to `0.0`?

File: src/org/usfirst/frc/team166/robot/commands/drive/DriveDistance.java [hunk 3]
```diff
@@ -12,7 +13,7 @@
 	double myDistance;
 
 	public DriveDistance(double speed, double distance) {
-		mySpeed = speed;
+		mySpeed = -speed;
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/commands/drive/DriveDistance.java]: Add a comment explaining why it's negated

File: src/org/usfirst/frc/team166/robot/subsystems/AimShooter.java [hunk 4]
```diff
@@ -48,7 +48,7 @@ private void updatePIDConstants() {
 	}
 
 	private double convertAngleToDisplacement(double angle) {
-		double displacement = ((angle - 45) * voltsPerDegree);
+		double displacement = ((angle - 40) * voltsPerDegree);
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/AimShooter.java]: What does this `40` mean? Is it the same `40` on line 60? If so, it should have been made a constant above to allow you to only have to change one place.

File: src/org/usfirst/frc/team166/robot/subsystems/Drive.java [hunk 5]
```diff
@@ -23,9 +23,9 @@
  */
 public class Drive extends Subsystem {
 
-	// final double distancePerPulse = 12 / 56320.0; // this makes perfect cents // no it doesn't it makes
-	// 2.1306818181...
-	final double distancePerPulse = 1.0 / 2500;
+	final double distancePerPulse = (Math.PI * 10.25) / 1024.0;
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/Drive.java]: Where did this come from? The spec sheet?

File: src/org/usfirst/frc/team166/robot/subsystems/Drive.java [hunk 6]
```diff
@@ -220,6 +226,16 @@ public void stop() {
 		tankDrive.tankDrive(0, 0);
 	}
 
+	public void setPIDConstants() {
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/Drive.java:228]: Function doesn't appear to be used - why is it in here?

File: src/org/usfirst/frc/team166/robot/subsystems/Drive.java [hunk 7]
```diff
@@ -240,7 +256,7 @@ public double getRightEncoder() {
 
 	public double getEncoderDistance() {
 		// returns the average of the distance traveled by the left and right encoders
-		return ((rightEncoder.getDistance() + leftEncoder.getDistance()) / 2);
+		return (-1 * (rightEncoder.getDistance()));
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/Drive.java:258]: You updated the code but not the comment

File: src/org/usfirst/frc/team166/robot/subsystems/Drive.java [hunk 8]
```diff
@@ -74,6 +74,7 @@
 
 	public Drive() {
 		initializeGear();
+		// setPIDConstants();
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/Drive.java]: Why add commented-out code? Either it should have been added in one commit, and then removed later on, in which case it's in the git history, or it wasn't worth adding in the first place.

File: src/org/usfirst/frc/team166/robot/subsystems/Drive.java [hunk 9]
```diff
@@ -163,10 +164,10 @@ public void spinRight() {
 		// spins the robot to the right at the minimum speed required to turn
 		if (isReversed) {
 			// if the robot is in reverse mode
-			tankDrive.tankDrive(spinSpeed, -spinSpeed, false);
+			tankDrive.tankDrive(spinSpeed * 3, -spinSpeed * 3, false);
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/Drive.java:166]: You use this `* 3` all over the place - it should be a constant above.

File: src/org/usfirst/frc/team166/robot/subsystems/Drive.java [hunk 10]
```diff
@@ -202,7 +204,11 @@ public void driveWithGyro(double left, double right) {
 		double power = 0;
 		power = (rightPower + leftPower) / 1.5;
 		if (Math.abs(right) > .05 || Math.abs(left) > .05) {
-			tankDrive.tankDrive(power - getGyroOffset(), power + getGyroOffset(), false);
+			if (isReversed) {
```
Comment (msoucy) [inline src/org/usfirst/frc/team166/robot/subsystems/Drive.java:206]: Why not just:

``` java
if(isReversed) {
    power *= -1;
}
tankDrive.tankDrive(power - getGyroOffset(), power + getGyroOffset(), false);
```

**Comment by msoucy** [comment]
- If you start a list with `- Foo` instead of `-Foo` it shows up with pretty bullets
- All of the autonomous classes are almost identical, only differing in certain numbers. Why not just have one class to maintain, and have the others use that?
- Has there really been 0 testing on this code? It shouldn't be merged in until we know that it does what's expected.

**Comment by atrimper** [comment]
Matt,
Most of the code has been tested (all but the seperate autonomous commands
for each spot on the field and the new spin speed). I will fix the mistakes
that we didn't fix before we do the final tests. I'm not sure if I said
that it wasn't tested, but nearly all of it was tested and is working.
-Andrew

**Comment by msoucy** [comment]
You did say that quite a bit wasn't tested, up at the top. What is the status of that?

**Comment by atrimper** [comment]
It would be great if we could merge this pull request even with the minor superficial errors so that we can continue working on this version on other laptops. I will clean up the other minor errors during testing so that we do not inadvertently create bugs. I changed a few of the simple errors.

**Comment by msoucy** [comment]
Part of the point of git is that you'd be able to pull from your own fork to fix it up later if needed. That said, I'll merge it to make it more accessible. It'll still need some work but it's at a state where it functions with the current robot.

**Pull request merged**